app/controllers/application_controller.rb

```diff
@@ -1,11 +1,19 @@
 class ApplicationController < ActionController::Base
   before_action :authenticate_user!
+  protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :load_and_authorize_resource
```

Comment:
**Optional**
- Well done using `load_and_authorize_resource` from the `cancan` gem so far. However, you're using it inside the application controller and this will make it apply to all other controllers, to be more accurate, I suggest to move it only in the posts controller so that it handles posts authorization only. 👍